src/EFCore.Relational/Extensions/RelationalDatabaseFacadeExtensions.cs

```diff
@@ -599,16 +598,9 @@ public static IDbContextTransaction BeginTransaction(this DatabaseFacade databas
             this DatabaseFacade databaseFacade,
             DbTransaction? transaction,
             Guid transactionId)
-        {
-            var transactionManager = GetTransactionManager(databaseFacade);
-
-            if (!(transactionManager is IRelationalTransactionManager relationalTransactionManager))
-            {
-                throw new InvalidOperationException(RelationalStrings.RelationalNotInUse);
-            }
-
-            return relationalTransactionManager.UseTransaction(transaction, transactionId);
-        }
+            => GetTransactionManager(databaseFacade) is not IRelationalTransactionManager relationalTransactionManager
+                ? throw new InvalidOperationException(RelationalStrings.RelationalNotInUse)
+                : relationalTransactionManager.UseTransaction(transaction, transactionId);
```

Comment:
nit: consider inversing logic

```suggestion
            => GetTransactionManager(databaseFacade) is IRelationalTransactionManager relationalTransactionManager
                ? relationalTransactionManager.UseTransaction(transaction, transactionId)
                : throw new InvalidOperationException(RelationalStrings.RelationalNotInUse);
```